src/worker_pool.rs

```diff
@@ -231,20 +270,60 @@ mod job_pool_tests {
         }
     }
 
-    // this test is ignored because it commits data to the db
+    fn get_all_tasks(conn: &PgConnection) -> Vec<Task> {
+        fang_tasks::table
+            .filter(fang_tasks::task_type.eq("worker_pool_test"))
+            .get_results::<Task>(conn)
+            .unwrap()
+    }
+
+    // Following tests ignored because they commit data to the db
+    #[test]
+    #[ignore]
+    fn tasks_are_finished_on_shutdown() {
+        let queue = Queue::new();
+
+        let mut worker_params = WorkerParams::new();
+        worker_params.set_retention_mode(RetentionMode::KeepAll);
+        let mut job_pool = WorkerPool::new_with_params(2, worker_params);
+
+        queue.push_task(&MyJob::new(100)).unwrap();
+        queue.push_task(&MyJob::new(200)).unwrap();
+
+        job_pool.start().unwrap();
+        thread::sleep(Duration::from_secs(1));
+        job_pool.shutdown().unwrap();
+
+        let tasks = get_all_tasks(&queue.connection);
+        // TODO - Replace with group_by when it's not nightly anymore
+        let new_tasks = tasks.iter().filter(|task| task.state == FangTaskState::New);
+        // let in_progress_tasks = tasks
```

Comment:
I think it doesn't test shutdown?

Reply:
I believe it does because I launch two tasks, those tasks start, create two new tasks. I shutdown while the original two tasks are running, this lets them finish; but the two newly created tasks never start, and shutdown then returns (it blocks till this happens). I then verify there are two completed, and two unstarted.

Reply:
Hmm, I see where you may be coming from. I've added an additional sleep so that if `shutdown` didn't kill the threads; there'd be enough time for them to create additional tasks, breaking the test.

Reply:
in the master branch, I added a new job type for this test